bots/core-logic/series-deactivator.js

```diff
@@ -0,0 +1,181 @@
+require('dotenv').config()
+const { ethers, utils } = require('ethers')
+
+const optionCatalogueAbi = require('../abi/OptionCatalogue.json')
+const liquidityPoolAbi = require('../abi/LiquidityPool.json')
+const beyondPricerAbi = require('../abi/BeyondPricer.json')
+const managerAbi = require('../abi/Manager.json')
+
+const seriesDeactivatorLogic = async (
+	signer,
+	optionCatalogueAddress,
+	liquidityPoolAddress,
+	beyondPricerAddress,
+	managerAddress,
+	collateralAssetAddress,
+	strikeAssetAddress,
+	underlyingAssetAddress
+) => {
+	const optionCatalogue = new ethers.Contract(optionCatalogueAddress, optionCatalogueAbi, signer)
+	const liquidityPool = new ethers.Contract(liquidityPoolAddress, liquidityPoolAbi, signer)
+	const beyondPricer = new ethers.Contract(beyondPricerAddress, beyondPricerAbi, signer)
+	const manager = new ethers.Contract(managerAddress, managerAbi, signer)
+
+	const minExpiryTime = (await liquidityPool.optionParams()).minExpiry.toNumber()
+	const expirations = await optionCatalogue.getExpirations()
+
+	const minDelta = 0.1
+	const maxDelta = 0.7
+
+	// create array to which we will add all Option types to change as we iterate
+	const totalInput = []
+
+	for (let i = 0; i < expirations.length; i++) {
+		console.log('i:', i)
+		console.log(new Date(expirations[i] * 1000))
+		const callStrikes = await optionCatalogue.getOptionDetails(expirations[i], false)
+		const putStrikes = await optionCatalogue.getOptionDetails(expirations[i], true)
+
+		console.log({
+			callStrikes: callStrikes.map(x => ethers.utils.formatEther(x)),
+			putStrikes: putStrikes.map(x => ethers.utils.formatEther(x))
+		})
+		if (expirations[i] < Date.now() / 1000 + minExpiryTime) {
```

Comment:
add a lightweight expiration checker so if theyre expired you can just ignore pretty much